src/Rescore/Query.php

```diff
@@ -50,7 +55,8 @@ public function toArray(): array
     /**
      * Sets rescoreQuery object.
      *
-     * @param BaseQuery|\Elastica\Query\AbstractQuery|string $rescoreQuery
+     * @param AbstractQuery|AbstractSuggest|array|BaseQuery|Collapse|string|Suggest|null $rescoreQuery
+     * @phpstan-param TCreateQueryArgs $rescoreQuery
```

Comment:
and here `TCreateQueryArgsForModifyingQueries` too.

Reply:
Fixed all of the above + changed `\Elastica\SearchableInterface::count()` too. Also, I renamed `TCreateQueryArgsForModifyingQueries` to `TCreateQueryArgsMatching`